lib/ansible/plugins/filter/core.py

```diff
@@ -254,42 +249,19 @@ def get_hash(data, hashtype='sha1'):
     return h.hexdigest()
 
 
-def get_encrypted_password(password, hashtype='sha512', salt=None):
+def get_encrypted_password(password, hashtype='sha512', salt=None, **settings):
```

Comment:
I'm not a fan of passing opaque data structures (```settings```) around.  I'd much rather that these parameters were explicitly listed. (In the other functions as well).